### ISSUE TITLE
consider release/acquire of the jobserver token around cache/dist-sccache requests

### ISSUE THREAD

**Issue by froydnj** [mozilla/sccache#688]
For parallel builds that are relying on remote caches (e.g. S3), letting some other process start while we are waiting on network I/O might be a net win.  Firefox's cache hits take ~0.1s (writes take ~0.2s), so it's not clear how much of a win this would be.

You could do a similar release/acquire around dist-sccache compilation requests, on the supposition that getting work into the scheduler's queue is the more important thing and that the build could be doing useful things for other files locally while the remote build machines are working.

**Comment by luser** [comment]
I've actually thought for a while that the jobserver model would break down with distributed builds. What you really want is two separate job pools: one for local work and one for remote work. The size of the former is easy enough to determine given the number of local CPU cores available, but the latter ought to be flexible based on how many build cores sccache reports are available.

I don't know that this is even really possible to do in GNU make, though. [Ninja has a `pools` feature](https://ninja-build.org/manual.html#ref_pool) which is similar to this. I believe the Chromium build uses it to limit the number of simultaneous linker invocations.

**Comment by froydnj** [comment]
Yeah, this is kind of trying to express the local/remote work dichotomy and/or async tasks, but very ad hoc.
